students/kristianjf/lesson09/mailroom_5.py

```diff
@@ -64,38 +88,39 @@ def thank_you(self, full_name='', donation_amount=int()):
         '''
         Collect, Format and Validate Full name
         '''
-        #full_name = ''
         while not full_name:
             full_name_input = input('Please enter full name of the recipient: ').split()
             full_name = self.name_response_valid(full_name_input)
         '''
         Process Full Name
         '''
-        ds_names = [name for name, donation in self.DATA_STRUCTURE]
+        ds_names = [name for name in self.donors]
 
         if full_name in ds_names:
-            full_name_index = ds_names.index(full_name)
             print(f'Found {full_name} in data_structure')
             while not donation_amount:
                 donation_amount = int(input('Please enter a donation amount: '))
-            self.DATA_STRUCTURE[full_name_index][1].append(donation_amount)
-            print(self.output_string().format(name=full_name, donation=donation_amount))
+            self.donors[full_name].donate(donation_amount)
+            print(self.donors[full_name].output_string().format(name=full_name, \
+            donation=donation_amount))
         else:
             while not donation_amount:
                 donation_amount = int(input('Please enter a donation amount: '))
-            self.DATA_STRUCTURE.append((full_name, [donation_amount]))
-            print(self.output_string().format(name=full_name, donation=donation_amount))
-        return self.output_string().format(name=full_name, donation=donation_amount)
+            self.add_donor(Donor(full_name, donation_amount))
+            print(self.donors[full_name].output_string().format(name=full_name, \
+            donation=donation_amount))
+        return self.donors[full_name].output_string().format(name=full_name, \
+        donation=donation_amount)
 
-    def list_donors(self):
+    def list_donors(self, **kwargs):
         '''
         If the user types ‘list’, show them a list of the donor names and re-prompt
         '''
-        for entry in self.DATA_STRUCTURE:
-            print(entry[0])
-        return [entry[0] for entry in self.DATA_STRUCTURE]
+        for name in self.donors:
```

Comment:
don't loop through to print, then again to return a list....

and if you are returning the list -- why print it here?